FirebaseVertexAI/Sources/GenerativeModel.swift

```diff
@@ -217,33 +217,31 @@ public final class GenerativeModel {
                                                         isStreaming: true,
                                                         options: requestOptions)
 
-    var responseIterator = generativeAIService.loadRequestStream(request: generateContentRequest)
-      .makeAsyncIterator()
+    let responseStream = generativeAIService.loadRequestStream(request: generateContentRequest)
+
     return AsyncThrowingStream {
-      let response: GenerateContentResponse?
       do {
-        response = try await responseIterator.next()
-      } catch {
-        throw GenerativeModel.generateContentError(from: error)
-      }
+        for try await response in responseStream {
+          // Check the prompt feedback to see if the prompt was blocked.
+          if response.promptFeedback?.blockReason != nil {
+            throw GenerateContentError.promptBlocked(response: response)
+          }
 
-      // The responseIterator will return `nil` when it's done.
-      guard let response = response else {
+          // If the stream ended early unexpectedly, throw an error.
+          if let finishReason = response.candidates.first?.finishReason, finishReason != .stop {
```

Comment:
Do we ever care about candidates other than the first one?

Reply:
Good point. We do as of late last week since requesting multiple candidates now works. I'll see if we can get aligned on how to handle the scenario. Off the top of my head there are a couple options we could pick:
1. No change, ensure the first candidate finished successfully (`finishReason == .stop`) or throw.
2. Ensure all candidates finished successfully or throw.
3. Ensure at least one candidate finished successfully or throw.

Reply:
Will handle this change in a future PR / release since it'll take some time.